modules/component/types/componentProps.ts

```diff
@@ -0,0 +1,23 @@
+import {Action} from '@action-land/core'
+
+/**
+ * Component typing signature
+ * 1. **iState**: Initial state required by component to exist.
```

Comment:
- `required by component to exist`?
- the component already has it's own init that guarantees it no? this is just the type of the returned value